File: reservation-service/src/main/java/com/yugendev/model/Reservation.java
package com.yugendev.model;

import io.micronaut.core.annotation.Introspected;
import io.micronaut.data.annotation.GeneratedValue;
import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.MappedEntity;
import lombok.*;

import java.time.LocalDateTime;
import java.util.Map;
import java.util.UUID;

@Data
@Introspected
@MappedEntity
public class Reservation {

    @Id
    @GeneratedValue
    private UUID reservationId;
    private UUID userId;
    private UUID showtimeId;
    private Map<String, Object> reservedSeats;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

}

File: reservation-service/internal/models/reservation.go
package models

import (
	"github.com/google/uuid"
	"time"
)

type Reservation struct {
	ReservationID uuid.UUID `gorm:"type:uuid;default:uuid_generate_v4();primary_key"`
	UserID        uuid.UUID `gorm:"type:uuid;not null"`
	ShowtimeID    uuid.UUID `gorm:"type:uuid;not null"`
	ReservedSeats string    `gorm:"type:jsonb;not null"`
	CreatedAt     time.Time `gorm:"default:CURRENT_TIMESTAMP"`
	UpdatedAt     time.Time `gorm:"default:CURRENT_TIMESTAMP"`
}

func (Reservation) TableName() string {
	return "reservations"
}
